server/webhook.go

```diff
@@ -18,56 +18,74 @@ import (
 	"github.com/mattermost/mattermost-server/model"
 )
 
-type JIRAWebhookIssue struct {
-	Self   string
-	Key    string
-	Fields struct {
-		Assignee    *jira.User
-		Reporter    *jira.User
-		Summary     string
-		Description string
-		Priority    *struct {
-			Id      string
-			Name    string
-			IconURL string
-		}
-		IssueType struct {
-			Name    string
-			IconURL string
-		}
-		Resolution *struct {
-			Id string
-		}
-		Status struct {
-			Id string
-		}
-		Labels []string
-	}
+const (
+	eventIssueCreated = uint64(1 << iota)
+	eventIssueDeleted
+	eventUnresolvedIssueDeleted
+	eventIssueAssigned
+	eventIssueReopened
+	eventIssueResolved
+	eventIssueUpdatedStatus
+	eventIssueUpdatePriority
+	eventIssueUpdateSummary
+	eventIssueUpdateDescription
+	eventIssueUpdateSprint
+	eventIssueUpdateRank
+	eventIssueUpdatedAttachment
+	eventIssueUpdatedLabels
+	eventCommentCreated
+	eventCommentDeleted
+	eventCommentUpdated
+	eventMax = iota
+)
+
+const maskLegacy = eventIssueCreated |
+	eventIssueReopened |
+	eventIssueResolved |
+	eventUnresolvedIssueDeleted
+
+const maskComments = eventCommentCreated |
+	eventCommentDeleted |
+	eventCommentUpdated
+
+const maskDefault = maskLegacy
+
+// The keys listed here can be used in the Webhook URL to control what events
+// are posted to Mattermost. A matching parameter with a non-empty value must
+// be added to turn on the event display.
+var eventParamMasks = map[string]uint64{
+	"ecomment_created": eventCommentCreated,         // new comments
+	"eistatus":         eventIssueUpdatedStatus,     // transitions like Done, In Progress
+	"eiassigned":       eventIssueAssigned,          // re-assigned
+	"eiprio":           eventIssueUpdatePriority,    // changes in priority
+	"uirename":         eventIssueUpdateSummary,     // issue renamed
+	"eiedit":           eventIssueUpdateDescription, // issue description edited
+	"eisprint":         eventIssueUpdateSprint,      // assigned to a different sprint
+	"eall_comments":    maskComments,                // all comment events
+	"eall":             ^(-1 << eventMax),           // all events
 }
 
 type JIRAWebhook struct {
-	WebhookEvent string
-	Issue        JIRAWebhookIssue
-	User         jira.User
-	Comment      struct {
-		Body         string
-		UpdateAuthor jira.User
-	}
-	ChangeLog struct {
+	WebhookEvent string       `json:"webhookEvent,omitempty"`
```

Comment:
standard to use lower snake case.

Reply:
These come from Jira

Reply:
Oh ya! Well I will take it up with Atlassian then :P